imports/ui/signup/signup.jsx

```diff
@@ -70,35 +53,38 @@ export class SignupPage extends Component {
     if (!this.isFormValid()) return
 
     const { info } = this.state
-
     // Copying all the input values and trimming them
     const signupInfo = Object.keys(info).reduce((all, curr) => {
       if (info[curr]) {
-        all[curr] = info[curr].trim()
+        all[curr] = info[curr]
       }
       return all
     }, {})
     const { submitSignupInfo } = actions
     this.props.dispatch(submitSignupInfo(signupInfo))
   }
-  toggleExistingBz = () => {
+
+  handleTermCheckbox = () => {
+    const { info } = this.state
     this.setState({
-      existingBz: !this.state.existingBz
+      termsAgreement: Date(),
```

Comment:
You should `new Date()` instead, so a real Date object is passed on and not a string

Reply:
I'd prefer to see have this captured as a Boolean and not a Date, but then recorded as a Date on the backend.